cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/KinesisConsumerIT.java

```diff
@@ -0,0 +1,177 @@
+import com.amazonaws.services.kinesis.model.GetRecordsResult;
+import com.amazonaws.services.kinesis.model.GetShardIteratorRequest;
+import com.amazonaws.services.kinesis.model.Shard;
+import com.amazonaws.services.kinesis.model.ShardIteratorType;
+import com.yugabyte.cdcsdk.testing.util.CdcsdkTestBase;
+import com.yugabyte.cdcsdk.testing.util.UtilStrings;
+
+/**
+ * Release test that verifies inserts, updates and deletes on a YugabyteDB
+ * database and
+ * writing to Amazon Kinesis
+ *
+ * @author Sumukh Phalgaonkar
+ */
+
+public class KinesisConsumerIT extends CdcsdkTestBase {
+
+    private static String stream_name = "dbserver1.public.test_table";
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        initializeContainers();
+
+        initHelpers(true, false, false);
+    }
+
+    @BeforeEach
+    public void beforeEachTest() throws Exception {
+        ybHelper.execute(UtilStrings.getCreateTableYBStmt(DEFAULT_TABLE_NAME));
+
+        cdcsdkContainer = TestHelper.getCdcsdkContainerForKinesisSink(ybHelper, "public." + DEFAULT_TABLE_NAME)
+                .withCreateContainerCmdModifier(cmd -> cmd.withUser(System.getenv("USERID")));
+        cdcsdkContainer.withNetwork(containerNetwork);
+
+        cdcsdkContainer.withFileSystemBind(System.getenv("AWS_SHARED_CREDENTIALS_FILE"),
+                "/home/jboss/.aws/credentials", BindMode.READ_ONLY);
+        cdcsdkContainer.start();
+
+        assertTrue(cdcsdkContainer.isRunning());
+
+    }
+
+    @AfterEach
+    public void dropTable() throws Exception {
+        ybHelper.execute(UtilStrings.getDropTableStmt(DEFAULT_TABLE_NAME));
+    }
+
+    @Test
+    public void automationOfKinesisAssertions() throws Exception {
+
+        int recordsInserted = 1;
+        String uuid = getUUID().substring(0, 8);
+        for (int i = 0; i < recordsInserted; ++i) {
+            ybHelper.execute(UtilStrings.getInsertStmt(DEFAULT_TABLE_NAME, i, uuid, "last_" + i, 23.45));
+        }
+        ybHelper.execute(UtilStrings.getUpdateStmt(DEFAULT_TABLE_NAME, 0, 25));
+
+        ybHelper.execute(UtilStrings.getDeleteStmt(DEFAULT_TABLE_NAME, 0));
+
+        Awaitility.await().atMost(Duration.ofSeconds(10));
+
+        List<String> expected_data = List.of(
+                "{\"id\":0,\"first_name\":\"" + uuid
+                        + "\",\"last_name\":\"last_0\",\"days_worked\":23.45,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"days_worked\":25.0,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"__deleted\":\"true\"}");
+
+        List<String> actual_data = new ArrayList<>();
+
+        AWSCredentials awsCredentials = new BasicSessionCredentials(System.getenv("AWS_ACCESS_KEY_ID"),
+                System.getenv("AWS_SECRET_ACCESS_KEY"),
+                System.getenv("AWS_SESSION_TOKEN"));
+
+        AmazonKinesisClient client = new AmazonKinesisClient(awsCredentials);
+
+        client.setEndpoint("https://kinesis.ap-south-1.amazonaws.com/");
```

Comment:
This also has to be a parameter from an env variable.

Reply:
Done